chain/messagepool/messagepool.go

```diff
@@ -139,9 +139,14 @@ type MessagePool struct {
 
 	keyCache *lru.Cache[address.Address, address.Address]
 
-	curTsLk sync.RWMutex // DO NOT LOCK INSIDE lk
+	stateLk sync.RWMutex
```

Comment:
Could you also had a similar comment as below describing this lock?

Reply:
done ty!!  4d458f119